src/VisualStudio/Core/SolutionExplorerShim/DiagnosticItem/DiagnosticItemProvider.cs

```diff
@@ -15,11 +16,14 @@ internal sealed class DiagnosticItemProvider : AttachedCollectionSourceProvider<
         [Import(typeof(AnalyzersCommandHandler))]
         private IAnalyzersCommandHandler _commandHandler = null;
 
+        [Import]
+        private IDiagnosticAnalyzerService _diagnosticAnalyzerService = null;
+
         protected override IAttachedCollectionSource CreateCollectionSource(AnalyzerItem item, string relationshipName)
         {
             if (relationshipName == KnownRelationships.Contains)
             {
-                return new DiagnosticItemSource(item, _commandHandler);
+                return new DiagnosticItemSource(item, _commandHandler, (DiagnosticAnalyzerService)_diagnosticAnalyzerService);
```

Comment:
We should also consider just adding the GetDiagnosticDescriptors(Analyzer) method to IDiagnosticAnalyzerService interface.